src/uct/sm/mm/mm_posix.c

```diff
@@ -31,11 +31,59 @@ static ucs_config_field_t uct_posix_pd_config_table[] = {
   {NULL}
 };
 
+static ucs_status_t uct_posix_test_mem(size_t length, int shm_fd)
+{
+    int *buf;
+    int chunk_size = 256 * 1024;
+    ucs_status_t status = UCS_OK;
+    size_t size_to_write, remaining, single_write, total_write = 0;
+
+    buf = ucs_malloc(chunk_size, "write buffer");
+    if (buf == NULL) {
+        ucs_error("Failed to allocate memory for testing space for backing file.");
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
+    }
+
+    memset(buf, 0, chunk_size);
+    if (lseek(shm_fd, 0, SEEK_SET) < 0) {
+        status = UCS_ERR_IO_ERROR;
+        goto out_free_buf;
+    }
+
+    remaining = length;
+    while (remaining > 0) {
+        size_to_write = ucs_min(remaining, chunk_size);
+        single_write = write(shm_fd, buf, size_to_write);
+        if (single_write < size_to_write) {
+            ucs_error("Failed to write %zu bytes. %m", size_to_write);
```

Comment:
are you sure errno would be set in case single_write is >=0 by < size_to_write? according to man page, "On error, -1 is returned, and errno is set appropriately".